controllers/cluster_controller.go

```diff
@@ -287,3 +296,57 @@ func splitMachineList(list *clusterv1.MachineList) (*clusterv1.MachineList, *clu
 	}
 	return controlplanes, nodes
 }
+
+func (r *ClusterReconciler) reconcileClusterStatus(ctx context.Context, cluster *clusterv1.Cluster) error {
+	if cluster.Status.ControlPlaneInitialized {
+		return nil
+	}
+
+	machines := &clusterv1.MachineList{}
+	err := r.Client.List(ctx, machines, client.InNamespace(cluster.Namespace))
+	if err != nil {
+		return err
+	}
+
+	for _, m := range machines.Items {
+		if util.IsControlPlaneMachine(&m) && m.Status.NodeRef != nil {
+			cluster.Status.ControlPlaneInitialized = true
+			return nil
+		}
+	}
+
+	return nil
+}
+
+// MachineToMachineSets is a handler.ToRequestsFunc to be used to enqeue requests for reconciliation
+// for Cluster to update its status.controlplaneInitialized field
+func (r *ClusterReconciler) MachineToCluster(o handler.MapObject) []ctrl.Request {
```

Comment:
```suggestion
func (r *ClusterReconciler) machineToCluster(o handler.MapObject) []ctrl.Request {
```

Reply:
i exported this to keep consistency with `MachineToMachineSets` and `MachineSetToDeployments`, whould we also un-export them too?